api/wasm/cpp/proxy_wasm_enums.h

```diff
@@ -59,3 +59,11 @@ enum class MetricType : EnumType {
   Gauge = 1,
   Histogram = 2,
 };
+enum class MetadataResult : EnumType {
```

Comment:
This is fine, but ultimately I think that we need to have `WasmResult` (or something like that) that we can use across the whole ABI surface, instead of having `MetadataResult`, `HttpResult`, etc. Similar to how syscalls share the error codes.

What do you think?

Reply:
Yes, that seems reasonable.  Let's create an issue and do that in another PR.  There are other cases which we can roll into that PR.

Reply:
https://github.com/envoyproxy/envoy-wasm/issues/104